compiler/resolution/generics.cpp

```diff
@@ -168,6 +168,25 @@ checkInfiniteWhereInstantiation(FnSymbol* fn) {
 }
 
 
+static Map<FnSymbol*,int> instantiationLimitMap;
+
+
+static bool trackInstantiationsForFn(FnSymbol* fn) {
+  ModuleSymbol* mod = fn->getModule();
+
+          // Don't count instantiations on internal modules
+  return (mod && mod->modTag != MOD_INTERNAL &&
+          // nor ones explicitly marked NO_INSTANTIATION_LIMIT.
+          !fn->hasFlag(FLAG_NO_INSTANTIATION_LIMIT) &&
+          // Nor ones that are compiler-generated (we could but
+          // this has caused problems for me in some cases and
+          // I'd prefer to assume compiler-generated functions
+          // won't result in infinitely recursive instantiations;
+          // to reproduce comment that part of the check out and try
+          // test/functions/resolution/instantiateMax/instMaxOKifNonrecursive.chpl
```

Comment:
I think we ought to understand what is going on in this case, before merging this PR.

Reply:
Also, need proper parenthesization and more periods in the comment.

Reply:
I believe what's happening is that when we're creating generic records, we have a special code path that's used to resolve their assignment functions, and that this code path doesn't insert the generics into the table:

https://github.com/chapel-lang/chapel/blob/ba4ada308ff162c0fdfead4e831cdb604182eade/compiler/resolution/resolveFunction.cpp#L1764-L1767

That said, I don't view this as a big issue, as I think it's unlikely that we're going to create infinitely-recursive compiler-generated functions, and even if we did, I think it's unlikely that we'd merge without realizing it.  And it has the side benefit of saving time by putting a bunch of compiler-generated stuff into the table and then removing it again (much as the pre-existing check against internal modules arguably did).